circadian/readers.py

```diff
@@ -1,49 +1,297 @@
+        df_dict[key] = df
+    return df_dict
+
+# %% ../nbs/api/05_readers.ipynb 14
+def load_csv(filepath: str, # full path to csv file to be loaded
+             metadata: Dict[str, str] = None, # metadata containing data_id, subject_id, or other_info
+             timestamp_col: str = None, # name of the column to be used as timestamp. If None, it is assumed that a `datetime` column exists
+             *args, # arguments to pass to pd.read_csv
+             **kwargs, # keyword arguments to pass to pd.read_csv
+             ):
+    "Create a dataframe from a csv containing wereable data"
+    # validate inputs
+    if not isinstance(filepath, str):
+        raise AttributeError("Filepath must be a string.")
+    if not isinstance(timestamp_col, str) and timestamp_col is not None:
+        raise AttributeError("Timestamp column must be a string.")
+    if metadata is not None:
+        WereableAccessor._validate_metadata(metadata)
+    # load csv
+    df = pd.read_csv(filepath, *args, **kwargs)
+    # create datetime column
+    if timestamp_col is not None:
+        df['datetime'] = pd.to_datetime(df[timestamp_col], unit='s')
+    if timestamp_col is None and 'datetime' not in df.columns:
+        raise AttributeError("CSV file must have a column named 'datetime' or a timestamp column must be provided.")
+    # add metadata
+    if metadata is not None:
+        df.wereable.add_metadata(metadata, inplace=True)
+    else:
+        df.wereable.add_metadata({'data_id': 'unknown', 'subject_id': 'unknown'}, inplace=True)
+    return df
+
+# %% ../nbs/api/05_readers.ipynb 17
+def distribute_stream_over_duration(start: pd.Timestamp, # start of the stream
+                                    end: pd.Timestamp, # end of the stream
+                                    value: float, # value of the stream
+                                    name: str, # name of the stream
+                                    resample_freq: str, # frequency to resample the stream to
+                                    ) -> pd.DataFrame: # dataframe with the distributed stream
+    "Distribute a wereable datapoint over its duration"
+    # validate inputs
+    if not isinstance(start, pd.Timestamp):
+        raise AttributeError("Start must be a pandas Timestamp.")
+    if not isinstance(end, pd.Timestamp):
+        raise AttributeError("End must be a pandas Timestamp.")
+    if not isinstance(value, (int, float)):
+        raise AttributeError("Value must be an integer or float.")
+    if not isinstance(resample_freq, str):
+        raise AttributeError("Resample frequency must be a string.")
+    # create dataframe
+    if pd.to_timedelta(resample_freq) > (end - start):
+        datetime = pd.to_datetime(start + (end - start) / 2, unit='s')
+        df = pd.DataFrame({'datetime': datetime, name: value}, index=[0])
+    else:
+        df = pd.DataFrame({'datetime': pd.date_range(start, end, freq=resample_freq)})
```

Comment:
- Create a time array that has the desired sampling frequency; 

e.g. `time_array = np.arange(0, T, delta_t)`
`values_array = np.zeros_likes(time_array)`

```for index, time in enumerated(time_array):
for wearable_data_point in all_data:

if time in interval(wearable_data_point.start, wearable_data_point.end):
value[index] = wearable_data_point.value / wearable_data_point.start / interval

Reply:
Take a look at this way vs. resample